src/careamics/file_io/write/get_func.py

```diff
@@ -1,14 +1,16 @@
 """Module to get write functions."""
 
 from pathlib import Path
-from typing import Protocol, Union
+from typing import Literal, Protocol
 
 from numpy.typing import NDArray
 
 from careamics.config.support import SupportedData
 
 from .tiff import write_tiff
 
+SupportedWriteType = Literal["tiff", "custom"]
```

Comment:
So far, we have used throughout the code a pattern of `enum.Enum` for the `Supported*`. 

I am not far enough in the review to have a comment yet, but do you see a reason to prefer a `Literal` to the `Enum`?

I'd argue that since literal are just annotations, Enum are the way to go to enforce discrete values in the code. They are unfortunately a bit more verbose...

Reply:
and if the `Literal` was used for mypy and its value never tested, then maybe an `Enum` is an overkill.

Reply:
So in my previous PR we spoke about using `Literal` for user facing functions. It is converted to `SupportedData` at some point before selecting the write function, although supported write types are actually a subset of `SupportedData` because `"array"` is not included.

Reply:
I will leave this as a `Literal` for now

Reply:
I don't have that good a memory. 😄 

Literals for user facing and Enum int he back end sounds good!